Tasks/Common/ios-signing-common/ios-signing-common.ts

```diff
@@ -225,20 +224,31 @@ export async function getProvisioningProfileUUID(provProfilePath: string) {
     })
     await plistTool.exec();
 
+    //use PlistBuddy to figure out the Name
+    let provProfileName: string;
+    plistTool = tl.tool(plist);
+    plistTool.arg(['-c', 'Print Name', tmpPlist]);
+    plistTool.on('stdout', function (data) {
+        if (data) {
+            provProfileName = data.toString();
+        }
+    })
+    await plistTool.exec();
+
     //delete the temporary plist file
     let deletePlistCommand: ToolRunner = tl.tool(tl.which('rm', true));
     deletePlistCommand.arg(['-f', tmpPlist]);
     await deletePlistCommand.exec();
 
-    if (provProfileUUID) {
+    if (provProfileUUID && provProfileName) {
         //copy the provisioning profile file to ~/Library/MobileDevice/Provisioning Profiles
         tl.mkdirP(userProvisioningProfilesPath); // Path may not exist if Xcode has not been run yet.
         let pathToProvProfile: string = getProvisioningProfilePath(provProfileUUID);
         let copyProvProfileCmd: ToolRunner = tl.tool(tl.which('cp', true));
         copyProvProfileCmd.arg(['-f', provProfilePath, pathToProvProfile]);
         await copyProvProfileCmd.exec();
 
-        return provProfileUUID;
+        return { provProfileUUID, provProfileName };
     } else {
         throw tl.loc('ProvProfileUUIDNotFound', provProfilePath);
```

Comment:
Is it possible for the UUID to be there but for the profile name to not be found?  Maybe we should update this error message.

Reply:
Apple generates provisioning profiles and I doubt they'd generate one without a Name. I made this best effort and added a warning just in case.